pkg/internal/controller/controller_test.go

```diff
@@ -169,6 +172,108 @@ var _ = Describe("controller", func() {
 			close(done)
 		}, 10.0)
 
+		It("should not error when source.channel is passed", func(done Done) {
+			eventCount := 0
+			// channel to be closed when event is processed
+			processed := make(chan struct{})
+			// source channel to be injected
+			ch := make(chan event.GenericEvent, 1)
+
+			ctx, cancel := context.WithCancel(context.TODO())
+			defer cancel()
+
+			// event to be sent to the channel
+			p := &corev1.Pod{
+				ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "bar"},
+			}
+			evt := event.GenericEvent{
+				Object: p,
+			}
+
+			ins := &source.Channel{Source: ch}
+			ins.DestBufferSize = 1
+			Expect(inject.StopChannelInto(ctx.Done(), ins)).To(BeTrue())
+
+			// send the event to the channel
+			ch <- evt
+
+			ctrl.startWatches = []watchDescription{{
+				src: ins,
+				handler: handler.Funcs{
+					GenericFunc: func(evt event.GenericEvent, q workqueue.RateLimitingInterface) {
+						defer GinkgoRecover()
+						eventCount++
+
+						if eventCount > 0 {
+							close(processed)
+						}
+					},
+				},
+			}}
+
+			go func() {
+				defer GinkgoRecover()
+				Expect(ctrl.Start(ctx)).To(Succeed())
+			}()
+
+			select {
+			case <-processed:
+				Expect(eventCount).To(BeEquivalentTo(1))
+			}
+
+			close(done)
+		})
+
+		It("should error when channel is passed as a source but stop channel is not injected", func(done Done) {
+			ch := make(chan event.GenericEvent)
+			ctx, cancel := context.WithCancel(context.TODO())
+			defer cancel()
+
+			ins := &source.Channel{Source: ch}
+			ctrl.startWatches = []watchDescription{{
+				src: ins,
+			}}
+
+			errChan := make(chan error)
+
+			go func() {
```

Comment:
We can simplify this by not starting in a goroutine that writes into a channel and then waiting for the channel and instead just start in the main goroutine or not?

Reply:
This is blocking. Starting in main routine is causing the test to time out.

Reply:
I know its blocking, my point was that its needlessly complicated to start it in a distinct goroutine.

```
errChan := make(chan error)
go func(){
  errChan <- someFunc()
}()
e := <-ErrChan
```

Is definitely semantically equivalent to
```
a := someFunc()
```

The test here works fine for me when changing it like described above and does not time out.